lightspark/requests/requester.py

```diff
@@ -103,18 +103,35 @@ def execute_graphql(
             result = r.json()
             if "errors" in result:
                 errors = result["errors"]
+                request_id = r.headers.get(
+                    "X-Request-ID", "Unknown"
+                )  # Get X-Request-ID from headers
+                logger.error(
+                    "GraphQL error occurred. Request ID: %s, Errors: %s",
+                    request_id,
+                    errors,
```

Comment:
I'm not totally sure whether this will serialize properly like this since it's an array. Might want to try in a test script.